docs/adr/adr_improved_frontend_dx.md

```diff
@@ -0,0 +1,55 @@
+# ADR: Improved Frontend Developer Experience
+Decision Deadline: As soon as possible
+
+Author: [Felipe](https://github.com/fhenrich33)
+
+## Status
+
+Proposed, Implementing
+
+## Context
+Frontend development experience in its current form is quite lacking. Imposes a lot of artificial difficulties, tooling doesn't work as expected, and adds friction to our FE contributors.
+
+We want to make FE development better.
+
+## Decision drivers
+
+1. We must make FE onboarding and development seamless.
```

Comment:
Seamless, as in independent of back end tooling?

Reply:
I have a bullet point for that, but yes. And also shouldn't be as wonky as it is now e.g. having a shell script to invoke Yarn, which for some reason doesn't work with my local installation.

Reply:
@fhenrich33 out of curiosity why do you need a shell script to invoke Yarn (and by extension) what kind of local installation set up do you have? 
Running `docker compose up` after managing the env vars did the trick (at least on my machine) so I am wondering what sort of errors you're getting.

Reply:
@jlin95 this script is used on our pre-commit setup. But since these live inside Docker, we don't have any of the expected linting, typing, and such from our local setup. As you have experienced here: https://github.com/boxwise/boxtribute/pull/1400#discussion_r1667623496